cmd/goat/bsky_prefs.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	appbsky "github.com/bluesky-social/indigo/api/bsky"
+
+	"github.com/urfave/cli/v2"
+)
+
+var cmdBskyPrefs = &cli.Command{
+	Name:  "prefs",
+	Usage: "sub-commands for preferences",
+	Flags: []cli.Flag{},
+	Subcommands: []*cli.Command{
+		&cli.Command{
+			Name:   "export",
+			Usage:  "dump preferences out as JSON",
+			Action: runBskyPrefsExport,
+		},
+		&cli.Command{
+			Name:      "import",
+			Usage:     "upload preferences from JSON file",
+			ArgsUsage: `<file>`,
+			Action:    runBskyPrefsImport,
+		},
+	},
+}
+
+func runBskyPrefsExport(cctx *cli.Context) error {
+	ctx := context.Background()
+
+	xrpcc, err := loadAuthClient(ctx)
+	if err == ErrNoAuthSession {
+		return fmt.Errorf("auth required, but not logged in")
+	} else if err != nil {
+		return err
+	}
+
+	// TODO: does this crash with unsupported preference types?
```

Comment:
it _shouldn't_

I think the client is more accepting than the server. I believe the PDS will reject requests for preferences not in the `app.bsky` namespace. Which also means that users migrating off of our PDSs shouldn't run into any unsupported prefs

Reply:
this is an indigo thing: the default de-serialization will throw an error if it encounters a `$type` which isn't a Lexicon schema known at compile-time.